geo/src/main/java/org/apache/carbondata/geo/GeoHashIndex.java

```diff
@@ -136,17 +139,21 @@ public void init(String indexName, Map<String, String> properties) throws Except
     }
     String GRID_SIZE = commonKey + "gridsize";
     String gridSize = properties.get(GRID_SIZE);
-    if (StringUtils.isEmpty(gridSize)) {
+    if (StringUtils.isEmpty(gridSize) ||
+        !Pattern.compile(POSITIVE_INTEGER_REGEX).matcher(gridSize).find()) {
       throw new MalformedCarbonCommandException(
-              String.format("%s property is invalid. %s property must be specified.",
-                      CarbonCommonConstants.SPATIAL_INDEX, GRID_SIZE));
+          String.format("%s property is invalid. %s property must be specified, "
+                  + "and the value must be positive integer.",
+              CarbonCommonConstants.SPATIAL_INDEX, GRID_SIZE));
     }
     String CONVERSION_RATIO = commonKey + "conversionratio";
     String conversionRatio = properties.get(CONVERSION_RATIO);
-    if (StringUtils.isEmpty(conversionRatio)) {
+    if (StringUtils.isEmpty(conversionRatio) ||
```

Comment:
Can move this validation to method and reuse

Reply:
Done